app/controllers/admin/merchants_controller.rb

```diff
@@ -6,6 +6,9 @@ def index
   def update
     merchant = Merchant.find(params[:id])
     merchant.enable_disable
+    if merchant.disabled
+      merchant.deactivate_items
+    end
```

Comment:
Add conditional and call method on the merchant that will deactivate the merchant's items when the merchant is disabled